src/migrateCalculatedMeasuresToActivePivot/migrateCalculatedMeasuresInWidgets.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  ContentRecord,
+  CubeName,
+  DataModel,
+  MdxSelect,
+  WidgetWithQueryState,
+} from "@activeviam/activeui-sdk";
+import { MdxString } from "@activeviam/mdx";
+import { migrateCalculatedMeasuresInMdx } from "./migrateCalculatedMeasuresInMdx";
+import { produce } from "immer";
+import _mapValues from "lodash/mapValues";
+
+export const migrateCalculatedMeasuresInWidgets = (
+  widgets: ContentRecord,
+  dataModel: DataModel,
+  namesOfCalculatedMeasurestoMigrate: string[],
+): {
+  cubeNames: { [measureName: string]: CubeName };
+  migratedWidgetsRecord: ContentRecord;
+} => {
+  // Create an empty object where each calculated measure used in a saved widget will be added as a key with its cubeName as a value.
+  const cubeNames: { [measureName: string]: CubeName } = {};
+
+  const migratedWidgetsRecord = produce(widgets, (draft) => {
+    draft.children!.content.children = _mapValues(
+      widgets.children!.content.children,
+      (widgetRecord) => {
+        const widgetState: WidgetWithQueryState<MdxSelect, "serialized"> =
+          JSON.parse(widgetRecord.entry.content);
+        const mdx: MdxString | undefined = widgetState.query.mdx;
```

Comment:
What you're doing with parsing/stringifying is perfectly correct, but now I am thinking that it would be nice to use the predefined `deserializeWidgetState`, `serializeWidgetState` instead: it does some of the work for us, and it allows to call `deserializeWidgetState` from `5.0` and `serializeWidgetState` from 5.1.